server/datastore/mysql/activities.go

```diff
@@ -183,3 +183,54 @@ func (ds *Datastore) MarkActivitiesAsStreamed(ctx context.Context, activityIDs [
 	}
 	return nil
 }
+
+func (ds *Datastore) ListHostUpcomingActivities(ctx context.Context, hostID uint, opt fleet.ListOptions) ([]*fleet.Activity, *fleet.PaginationMetadata, error) {
+	// TODO(mna): pending feedback on https://github.com/fleetdm/fleet/pull/15931/files#r1455916027
```

Comment:
https://github.com/fleetdm/fleet/pull/15931/files#r1455916027

May require following up with Noah/product.

Reply:
if we need to include an entity that doesn't have an uuid, we could just add a column to that table, right?

Reply:
> if we need to include an entity that doesn't have an uuid, we could just add a column to that table, right?

Absolutely, yes, my question is more because the `upcoming` API was documented as returning an `id` (and not a `uuid`), but we don't have a relevant `id` to return in this case (the upcoming queue is currently just the pending scripts, but it could grow to include other pending actions, and they won't have a unique monotonic `id`, but they _will_ have a unique `uuid`).

Reply:
sorry, I should have clarified! this was a separate question of mine to make sure I understood the plan for these tables. Thanks!